webClient/src/app/browser-panel/browser-panel.component.ts

```diff
@@ -141,78 +158,39 @@ export class BrowserPanelComponent implements OnInit {
         this.connection.ftaWs.fopen(side, pathToOpen, mode, onOpen);
     }
 
-    // downloadFile(side: FTASide, filePath: string, dataHandler: (chunk: ArrayBuffer) => void, completeHandler: (err: any) => void): void {
-    //     this.fopen(side, filePath, FTAFileMode.read, (err, streamId) => {
-    //         this.connection.ftaWs.onBinaryData(streamId, (binData: FTABinaryData) => {
-    //             dataHandler(binData.data);
-    //         }, () => {
-    //             completeHandler(null);
-    //         });
-    //     });
-    // }
-
     sendTo(): void {
 
     }
 
+    showUploadModal(): void {
+        this.uploadModalVisible = true;
+    }
+
+    closeUploadModal(): void {
+        this.uploadModalVisible = false;
+    }
+
+    getSelectedDirectory(): string {
+        // console.log('Getting the correct directory');
+        if (this.isFolder(this.treeSelectedNode)) {
+            return this.selectedPath;
+        } else {
+            return this.getPathFromRoot(this.treeSelectedNode.parent);
+        }
+    }
+
     saveAs(): void {
-        let uri = ZoweZLUX.uriBroker.unixFileUri('contents', this.leftPath.slice(1), undefined, undefined, undefined, true);
-        console.log(uri);
-        const tokens = this.leftPath.split('/');
+        const tokens = this.selectedPath.split('/');
         const filename = tokens[tokens.length - 1];
         console.log('saveAs filename=' + filename);
         const a = document.createElement('a');
-        a.href = uri;
+        // TODO replace this call with a uri from the URI Broker
+        a.href = '/unixfile/contents' + this.selectedPath;
+        console.log(a.href);
```

Comment:
You should never do console.log in a component, because you have the ability to easily make use of our logger instead.
In your constructor, do:
```constructor(@Inject(Angular2InjectionTokens.LOGGER) private log: ZLUX.ComponentLogger, foo)```
Then, replace this console.log with a this.log.debug. This way, people dont need to see all your debug logs.